src/helpers.ts

```diff
@@ -68,11 +68,22 @@ export async function requiredStatusChecksForBranch(
     })
   ).data
 
+  var checksFromBranchProtection = []
+
+  const rules = (
+    await octokit.rest.repos.getBranchRules({
+      ...github.context.repo,
+      branch: branchName,
+    })
+  ).data
+
+  const checksFromRules = rules.filter(rule => rule?.parameters?.required_status_checks)
```

Comment:
this doesn't seem to be working probably because i don't fully understand how object chaining works
```
Property 'parameters' does not exist on type
```

what's the best way to filter an array of objects by a specific key in typescript?